umake/ui/cli/__init__.py

```diff
@@ -157,6 +158,41 @@ def mangle_args_for_default_framework(args):
     return result_args
 
 
+def get_frameworks(args):
+    categories = list_frameworks()
+    print_result = str()
+
+    if args.list or args.list_available:
+        # Sort the categories to prevent a random list at each new program execution
+        for category in sorted(categories, key=lambda cat: cat["category_name"]):
+            print_result += "{}: {}\n".format(category["category_name"], category["category_description"])
+
+            # Sort the frameworks to prevent a random list at each new program execution
+            for framework in sorted(category["frameworks"], key=lambda fram: fram["framework_name"]):
+                if args.list_available:
+                    if not framework["is_installable"]:
+                        continue
+
+                print_result += "\t{}: {} ".format(framework["framework_name"], framework["framework_description"])
+
+                if not framework["is_installable"]:
+                    print_result += "({})".format(_('not installable on this machine'))
```

Comment:
Please use here:
```
print_result = _("{} ({}))".format(print_result, _('not installable on this machine'))
```

This way:
* the string is now translatable
* it's better for i18n, because we can switch between different kind of ordering, depending on the locale (as the first part is internationalized, and not just appended)

So, check the same for the other "print_result +=" ones as well.
I think the only place it's valid to use this is when you append a new element to the list (so after a \n)